java/jdbc/src/main/java/com/flipkart/vitess/jdbc/VitessConnection.java

```diff
@@ -247,10 +249,8 @@ public void setReadOnly(boolean readOnly) throws SQLException {
                 Constants.SQLExceptionMessages.METHOD_CALLED_ON_OPEN_TRANSACTION);
         }
 
-        if (readOnly) {
-            throw new SQLFeatureNotSupportedException(Constants.SQLExceptionMessages.READ_ONLY);
-        }
-        this.readOnly = false;
+        // TODO implement read only properly
+        this.readOnly = readOnly;
```

Comment:
If it's easy to do, just do it? :)

But I agree that it's a weird feature in the API. If we merge this as is, our implementation will basically become incorrect as it never enforces readonly. Is this going to cause any actual harm? My gut feeling is "no, it doesn't", but I'm lacking the experience with JDBC here.

@tirsen I would say this is your call. We can merge as is or you can actually add the enforcement of it. Both works for me.